src/main/java/org/magicdgs/http/jsr203/HttpPath.java

```diff
@@ -239,12 +240,9 @@ private boolean endsWith(final byte[] other) {
             }
         }
 
-        // TODO: this codepath is not exercised at all because we always have
-        // TODO: absolute paths, and thus the first statement is always true
-        // TODO: this will change in the future (https://github.com/magicDGS/jsr203-http/issues/12)
         // final check for name boundary
-        return other[olast + 1] == HttpUtils.HTTP_PATH_SEPARATOR_CHAR
-                || last == -1 || this.normalizedPath[last] == HttpUtils.HTTP_PATH_SEPARATOR_CHAR;
+        return last == -1 || this.normalizedPath[last] == HttpUtils.HTTP_PATH_SEPARATOR_CHAR ||
```

Comment:
This is too convoluted - should clean the implementation and fix it (it is wrong). Also, requires more tests